tests/BookReader.test.js

```diff
@@ -18,3 +37,88 @@ test('has registered view type selected events', () => {
 test('has registered fullscreen toggle event', () => {
   expect(BookReader.eventNames.fullscreenToggled).toBeTruthy();
 });
+
+test('checks cookie when initParams called', () => {
+  br.getResumeValue = jest.fn(() => 15);
+  br.urlReadFragment = jest.fn(() => '');
+
+  const params = br.initParams();
+  expect(br.getResumeValue).toHaveBeenCalledTimes(1);
+  expect(params.init).toBe(true);
+  expect(params.index).toBe(15);
+  expect(params.fragmentChange).toBe(true);
+});
+
+test('does not check cookie when initParams called', () => {
+  br.getResumeValue = jest.fn(() => null);
+  br.urlReadFragment = jest.fn(() => '');
+  br.options.enablePageResume = false;
+
+  const params = br.initParams();
+  expect(br.getResumeValue).toHaveBeenCalledTimes(0);
+  expect(params.init).toBe(true);
+  expect(params.index).toBe(0);
+  expect(params.fragmentChange).toBe(false);
+});
+
+test('overrides cookie if page fragment when InitParams called', () => {
+  br.getResumeValue = jest.fn(() => 15);
+  br.urlReadFragment = jest.fn(() => 'page/n4');
+  br.options.enablePageResume = true;
+
+  const params = br.initParams();
+  expect(br.getResumeValue).toHaveBeenCalledTimes(1);
+  expect(params.init).toBe(true);
+  expect(params.index).toBe(4);
+  expect(params.fragmentChange).toBe(true);
+});
+
+test('sets prevReadMode when init called', () => {
+  br.init();
+  expect(br.prevReadMode).toBeTruthy();
+});
+
+test('sets prevPageMode if initial mode is thumb', () => {
+  br.urlReadFragment = jest.fn(() => 'page/n4/mode/thumb');
+
+  br.init();
+  expect(br.prevReadMode).toBe(BookReader.constMode1up);
+});
+
+test('calls switchMode with init option when init called', () => {
+  br.switchMode = jest.fn();
+
+  br.init();
+  expect(br.switchMode.mock.calls[0][1])
+    .toHaveProperty('init', true);
+});
+
+test('has suppressFragmentChange true when init with no input', () => {
+  br.getResumeValue = jest.fn(() => null);
+  br.urlReadFragment = jest.fn(() => '');
+  br.switchMode = jest.fn();
+
+  br.init();
+  expect(br.switchMode.mock.calls[0][1])
+    .toHaveProperty('suppressFragmentChange', true);
+});
+
+test('has suppressFragmentChange false when init with cookie', () => {
+  br.getResumeValue = jest.fn(() => 5);
+  br.urlReadFragment = jest.fn(() => '');
+  br.switchMode = jest.fn();
+
+  br.init();
+  expect(br.switchMode.mock.calls[0][1])
+    .toHaveProperty('suppressFragmentChange', false);
+});
+
+test('has suppressFragmentChange false when init with fragment', () => {
+  br.getResumeValue = jest.fn(() => null);
```

Comment:
q: does this mean that when a user goes to a book with url fragment: `/page/22/mode/2up` - you allow the `fragmentChange` event on init and that sets off its respective cascading changes/updates?

Reply:
yes